addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/models/NeeoDeviceType.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.neeo.internal.models;
+
+import java.util.Objects;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Defines the various NEEO device types. Many of these don't work yet (like CLIMA)
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+public enum NeeoDeviceType {
+    EXCLUDE(""),
```

Comment:
This was actually a great comment but for the wrong reasons.  Because NEEO is still immature, things are rapidly changing and I didn't want to limit the addon to what was known at the time of writing.  So I wanted the users to be able to enter and use any string (representing a device type or button) that NEEO supports.  However, I lost sight of that goal somewhere and created this class as an enum.  This should have really been a helper class (not enum) and accepts whatever type of input the user enters and passes it along to the NEEO brain.  I'm rewriting this class (and the buttons class - special buttons class is fine since that's get's converted to buttons) to reflect that goal